src/Data/PreciseDateTime.purs

```diff
@@ -92,8 +93,9 @@ adjust pd (PreciseDateTime dt ns) = do
     nsPrecDur = toNanoseconds pd
     nsPrecDurInt = unPreciseDuration nsPrecDur
     msPrecDur = toMilliseconds nsPrecDur
-    msPrecDurInt = unPreciseDuration msPrecDur
-    roundTripDurInt = unPreciseDuration <<< toNanoseconds $ msPrecDur
+    -- Truncate milliseconds to remove fractional nanoseconds.
+    msPrecDurInt = truncated $ unPreciseDuration msPrecDur
+    roundTripDurInt = unPreciseDuration <<< toNanoseconds $ Milliseconds msPrecDurInt
```

Comment:
Should `toNanoseconds` just handle this?

Reply:
Or rather it somehow be transparent and handled in `Data.Time.PreciseDuration`.

Reply:
I think not... this was kind of tricky to get right, `roundTripDurInt` must be `nsPrecDurInt` with everything smaller than milliseconds removed, so that `diffDurInt` below is nonzero.